src/hal/src/lib.rs

```diff
@@ -55,6 +56,9 @@ pub mod queue;
 pub mod range;
 pub mod window;
 
+#[doc(hidden)]
+pub mod auxil;
```

Comment:
actually, it appears to me that `backend` has a very similar goal. Should we move the fast hashmap into `backend`?